### PR TITLE
adding nsls2-collection & nsls2-analysis metadata

### PR DESCRIPTION
**Pull request by kbeeperez**
implements: https://github.com/nsls-ii-forge/collection-feedstock
https://github.com/nsls-ii-forge/nsls2-analysis-feedstock
https://github.com/nsls-ii-forge/analysis-feedstock

Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [ ] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [ ] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Review by mrakitin — CHANGES_REQUESTED** [review]
@kbeeperez, it feels a bit wrong.

We need the following 2 recipes:
- `nsls2-analysis` that would contain the combination of `run` requirements from these feedstocks' recipes and won't have the `analysis` dependency:
  - https://github.com/nsls-ii-forge/analysis-feedstock
  - https://github.com/nsls-ii-forge/nsls2-analysis-feedstock
- `nsls2-collection` that would be a copy of the recipe from https://github.com/nsls-ii-forge/collection-feedstock, and will depend on the `nsls2-analysis` metapackage from above.

File: recipes/nsls2-analysis/meta.yaml
Comment (mrakitin) [inline recipes/nsls2-analysis/meta.yaml:35]: The `dask_pin` variable is not defined. That should be added on the top of the file next to other Jinja2 vars.

File: recipes/nsls2-analysis/meta.yaml
Comment (mrakitin) [inline recipes/nsls2-analysis/meta.yaml:40]:
```suggestion
    # - edrixs
```

File: recipes/nsls2-analysis/meta.yaml
Comment (mrakitin) [inline recipes/nsls2-analysis/meta.yaml:55]: `mkl_pin` should be defined too.

File: recipes/nsls2-analysis/meta.yaml
Comment (mrakitin) [inline recipes/nsls2-analysis/meta.yaml:142]:
```suggestion
```

File: recipes/nsls2-analysis/meta.yaml
Comment (mrakitin) [inline recipes/nsls2-analysis/meta.yaml:195]:
```suggestion
```

File: recipes/nsls2-analysis/meta.yaml
Comment (mrakitin) [inline recipes/nsls2-analysis/meta.yaml:205]: The `test.xlsx` and `test.xls` files are missing. Please add them to the `recipe` dirs from the nsls2forge feedstocks.

File: recipes/nsls2-collection/meta.yaml
Comment (mrakitin) [inline recipes/nsls2-collection/meta.yaml:19]: This is to pick up the build number `0` for the `nsls2-analysis` recipe above.
```suggestion
    - nsls2-analysis ==2021C2.0=*_0
```

File: recipes/nsls2-collection/meta.yaml
Comment (mrakitin) [inline recipes/nsls2-collection/meta.yaml:25]:
```suggestion
    - nslsii >=0.1.3
```

File: recipes/nsls2-collection/meta.yaml
Comment (mrakitin) [inline recipes/nsls2-collection/meta.yaml:35]: This is already a dependency of `nsls2-analysis`.
```suggestion
```

File: recipes/nsls2-collection/meta.yaml
Comment (mrakitin) [inline recipes/nsls2-collection/meta.yaml:38]: This is already a dependency of `nsls2-analysis`.
```suggestion
```

File: recipes/nsls2-collection/meta.yaml
Comment (mrakitin) [inline recipes/nsls2-collection/meta.yaml:42]: This is already a dependency of `nsls2-analysis`.
```suggestion
```

File: recipes/nsls2-collection/meta.yaml
Comment (mrakitin) [inline recipes/nsls2-collection/meta.yaml:53]:
```suggestion
```

File: recipes/nsls2-collection/meta.yaml
Comment (mrakitin) [inline recipes/nsls2-collection/meta.yaml:112]:
```suggestion
```

File: recipes/nsls2-collection/meta.yaml
Comment (mrakitin) [inline recipes/nsls2-collection/meta.yaml:122]: Please add these missing files for tests.

File: recipes/nsls2-collection/yum_requirements.txt
Comment (mrakitin) [inline recipes/nsls2-collection/yum_requirements.txt:1]: The same file should be created for the `nsls2-analysis` recipe.

**Review by mrakitin — CHANGES_REQUESTED** [review]
@kbeeperez, here are a few suggestions after my review.

We need `xraylarch` (https://github.com/conda-forge/staged-recipes/pull/15706) to exist before we can run a new build. The other packages should already be in conda-forge.

File: recipes/nsls2-analysis/meta.yaml
Comment (mrakitin) [inline recipes/nsls2-analysis/meta.yaml:128]:
```suggestion
    - suitcase-mongo  # >=0.3.0 # remove the pin until https://github.com/conda-forge/suitcase-mongo-feedstock/pull/5 is merged
```

File: recipes/nsls2-analysis/meta.yaml
Comment (mrakitin) [inline recipes/nsls2-analysis/meta.yaml:143]:
```suggestion
    # - xraylarch  # commented out as https://github.com/conda-forge/staged-recipes/pull/15706 is not merged yet
```

**Review by mrakitin — COMMENTED** [review]
A couple of work-around suggestions before the awaited PRs are merged.

File: recipes/nsls2-analysis/meta.yaml
Comment (mrakitin) [inline recipes/nsls2-analysis/meta.yaml:44]:
```suggestion
    # - edrixs
```

File: recipes/nsls2-analysis/meta.yaml
Comment (mrakitin) [inline recipes/nsls2-analysis/meta.yaml:61]:
```suggestion
    - ipympl >=0.1.1
```

File: recipes/nsls2-analysis/meta.yaml
Comment (mrakitin) [inline recipes/nsls2-analysis/meta.yaml:63]:
```suggestion
    - ipywidgets >=7.2.1
```

**Review by mrakitin — COMMENTED** [review]
A few suggestions to relax the hard pins that don't exist for Python 3.7+.

**Comment by mrakitin** [comment]
I ran a build with `mamba` locally, and it uncovered a missing dependency of `pyxdameraulevenshtein` for Python 3.7+:
image


I submitted a PR to get the missing versions via https://github.com/conda-forge/pyxdameraulevenshtein-feedstock/pull/25 (we did it for our [`nsls2forge` feedstock](https://github.com/nsls-ii-forge/pyxdameraulevenshtein-feedstock/blob/587b50ff408e877cd2fdbb94d6dc1e514aa22a31/recipe/meta.yaml#L16) that way too, and it worked fine).

File: recipes/nsls2-analysis/meta.yaml
Comment (mrakitin) [inline recipes/nsls2-analysis/meta.yaml:141]:
```suggestion
    # - xpdan  # Blocker: https://github.com/conda-forge/pyxdameraulevenshtein-feedstock/pull/25
```

File: recipes/nsls2-collection/meta.yaml
Comment (mrakitin) [inline recipes/nsls2-collection/meta.yaml:38]:
```suggestion
    # - xpdacq  # Blocker: https://github.com/conda-forge/pyxdameraulevenshtein-feedstock/pull/25
```

**Review by mrakitin — COMMENTED** [review]
A couple of suggestions to have a successful build. Let's exclude `xpdan`/`xpdacq` for now - they can be added to the feedstock once the blocker PR is merged.

**Comment by mrakitin** [comment]
@dopplershift, we have a couple of blockers, but we commented out the corresponding dependencies for now, that can be added later in the feedstocks. Are you OK with that approach?

Blockers:
- https://github.com/conda-forge/pyxdameraulevenshtein-feedstock/pull/25
- https://github.com/conda-forge/staged-recipes/pull/15706

File: recipes/nsls2-analysis/meta.yaml
Comment (mrakitin) [inline recipes/nsls2-analysis/meta.yaml:141]: The https://github.com/conda-forge/pyxdameraulevenshtein-feedstock/pull/25 is merged now, the dependency can be added back.
```suggestion
    - xpdan
```

File: recipes/nsls2-analysis/LICENSE
Comment (mrakitin) [inline recipes/nsls2-analysis/LICENSE:13]: Could you please use a better-formatted license from https://github.com/bluesky/bluesky/blob/master/LICENSE and change the year to `2021` for both `nsls2-analysis` and `nsls2-collection` recipes?

File: recipes/nsls2-analysis/meta.yaml
Comment (mrakitin) [inline recipes/nsls2-analysis/meta.yaml:143]: Suggesting adding `xrt` to the analysis metapackage, as we may need it along with `srwpy` and other simulation packages.
```suggestion
    # - xraylarch  # commented out as https://github.com/conda-forge/staged-recipes/pull/15706 is not merged yet
    - xrt
```

File: recipes/nsls2-analysis/meta.yaml
Comment (mrakitin) [inline recipes/nsls2-analysis/meta.yaml:121]: One more simulation package:
```suggestion
    - scipy
    - shadow3
```

File: recipes/nsls2-analysis/meta.yaml
Comment (mrakitin) [inline recipes/nsls2-analysis/meta.yaml:99]:
```suggestion
    - pillow
    - pocl  # needed by pyopencl, used by the `xrt` package
```

File: recipes/nsls2-collection/meta.yaml
Comment (mrakitin) [inline recipes/nsls2-collection/meta.yaml:38]:
```suggestion
    - xpdacq
```

**Review by mrakitin — CHANGES_REQUESTED** [review]
Pending https://github.com/conda-forge/xray-vision-feedstock/pull/14. Once merged, my suggestions can be committed to trigger a new build with the updated dependencies.

**Comment by mrakitin** [comment]
@kbeeperez, the `xray-vision` with the fixed dependencies (done via https://github.com/conda-forge/xray-vision-feedstock/pull/14) is available now. Please commit my suggestions.

File: recipes/nsls2-collection/meta.yaml
Comment (mrakitin) [inline recipes/nsls2-collection/meta.yaml:68]: This import test was intended to be run against the `srw` package coming from https://github.com/nsls-ii-forge/srw-feedstock/, which we don't package in conda-forge, where `srwpy` (see below) should be the right one.
```suggestion
```

**Comment by mrakitin** [comment]
Success!

@conda-forge/staged-recipes, I agree to be a maintainer of these feedstocks. Could we please have them metged soon? Thank you!

**Comment by mrakitin** [comment]
@dopplershift, @ocefpaf, would you be able to review/merge it today? This PR is one of the final ones that NSLS-II needs for the deployment (besides the awaiting https://github.com/conda-forge/staged-recipes/pull/15706).

Thanks a lot!

File: recipes/nsls2-collection/meta.yaml
Comment (mrakitin) [inline recipes/nsls2-collection/meta.yaml:26]: @kbeeperez, let's add one more package from our LCLS friends (https://github.com/slaclab/pydm).
```suggestion
    - pexpect
    - pydm
```

**Review by leofang — COMMENTED** [review]
Is it possible to move *.xls and *.xlsx to somewhere else (say a GH repo) and download them at the test stage? This way we can avoid making them be a part of `staged-recipes`'s git history.

**Comment by mrakitin** [comment]
Hi @leofang! That's a reasonable concern. They can be added at a later time into the feedstock. The files are already in https://github.com/nsls-ii-forge/collection-feedstock/tree/master/recipe, so they can be taken from there. However, we have to do "git surgery" on the commits to this repo to properly exclude. Or squash-merge would take care of it. Would it work that way?

**Comment by leofang** [comment]
My guess is we can just squash and merge, but let's wait for someone from @conda-forge/staged-recipes to decide 😄 Maybe they'd just be ok with it.

**Comment by leofang** [comment]
Hi @carterbox any chance you have some bandwidth to help our NSLS-II friends here? 🙂

**Comment by mrakitin** [comment]
We currently have a workaround using the environment .yml files, so these metapackages may not be needed at this time.

File: recipes/nsls2-analysis/meta.yaml
Comment (carterbox) [inline recipes/nsls2-analysis/meta.yaml:20]: Since this is a metapackage, it shouldn't have a host/build section?

File: recipes/nsls2-collection/meta.yaml
Comment (carterbox) [inline recipes/nsls2-collection/meta.yaml:17]: No host/build for metapackages?

**Review by carterbox — CHANGES_REQUESTED** [review]
Is there a reason why these meta packages cannot be `noarch`? All conda-forge packages that can be noarch must be marked as `noarch` because it prevents unnecessary builds/migrations.

**Comment by mrakitin** [comment]
I don't think it's needed. We have a way to specify all those packages via an environment file. Can be closed.

**Pull request closed without merging**
